internal/service/transfer/connector_data_source.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package transfer
+
+import (
+	"context"
+
+	"github.com/YakDriver/regexache"
+	"github.com/aws/aws-sdk-go-v2/service/transfer"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework/flex"
+	fwtypes "github.com/hashicorp/terraform-provider-aws/internal/framework/types"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @FrameworkDataSource(name="Connector")
+func newDataSourceConnector(context.Context) (datasource.DataSourceWithConfigure, error) {
+	return &dataSourceConnector{}, nil
+}
+
+const (
+	DSNameConnector = "Connector Data Source"
+)
+
+type dataSourceConnector struct {
+	framework.DataSourceWithConfigure
+}
+
+func (d *dataSourceConnector) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) { // nosemgrep:ci.meta-in-func-name
+	resp.TypeName = "aws_transfer_connector"
+}
+
+func (d *dataSourceConnector) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			// Connector object was expanded
+			"access_role": schema.StringAttribute{
+				Computed: true,
+			},
+			names.AttrARN: schema.StringAttribute{
+				Computed: true,
+			},
+			"as2_config": schema.ListAttribute{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[dsAs2Config](ctx),
+				Computed:   true,
+			},
+			names.AttrID: schema.StringAttribute{
+				CustomType: fwtypes.RegexpType,
+				Required:   true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexache.MustCompile(`c-([0-9a-f]{17})`),
+						""),
+					stringvalidator.LengthAtMost(19),
+					stringvalidator.LengthAtLeast(19),
+				},
+			},
+			"logging_role": schema.StringAttribute{
+				Computed: true,
+			},
+			"security_policy_name": schema.StringAttribute{
+				Computed: true,
+			},
+			"service_managed_egress_ip_addresses": schema.ListAttribute{
+				CustomType: fwtypes.ListOfStringType,
+				Computed:   true,
+			},
+			"sftp_config": schema.ListAttribute{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[dsSftpConfig](ctx),
+				Computed:   true,
+			},
+			"tags": schema.ListAttribute{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[dsTags](ctx),
+				Computed:   true,
+			},
```

Comment:
`tags` are a special case and are handles as maps

```suggestion
			names.AttrTags: tftags.TagsAttributeComputedOnly(),
```